json-schema/json-schema/src/main/scala/endpoints/algebra/JsonSchemas.scala

```diff
@@ -72,6 +72,15 @@ trait JsonSchemas {
     */
   type Tagged[A] <: JsonSchema[A]
 
+  /** A JSON schema for enumerations, i.e. types that have a restricted set of values. */
+  type Enum[A] <: JsonSchema[A]
+
+  /** Promotes a schema to an enumeration and converts between enum constants and JSON strings.
+    * Decoding fails if the input string does not match the encoded values of any of the possible values.
+    * Encoding does never fail, even if the value is not contained in the set of possible values.
+    * */
+  def enumeration[A](values: Seq[A], encode: A => String)(implicit tpe: JsonSchema[String]): Enum[A]
```

Comment:
One last thing that comes to my mind is that we should use two parameter lists to improve type inference and support the following syntax:

~~~ scala
enumeration(Seq(Foo, Bar, Baz)) {
  case Foo => "foo"
  case Bar => "bar"
  case Baz => "baz"
}
~~~

Reply:
Sounds good. All done from my side.